src/schemathesis/specs/openapi/stateful/links.py

```diff
@@ -0,0 +1,128 @@
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Tuple
+
+import hypothesis.strategies as st
+import requests
+from hypothesis.stateful import Bundle
+
+from ....utils import NOT_SET
+from .. import expressions
+from ..constants import LOCATION_TO_CONTAINER
+from ..links import get_target_endpoints
+from ..utils import expand_status_code
+
+if TYPE_CHECKING:
+    from ....models import Case, Endpoint
+
+FilterFunction = Callable[[Tuple["Case", requests.Response]], bool]
+
+
+def apply(
+    endpoint: "Endpoint",
+    bundles: Dict[str, Bundle],
+    sources: Dict[str, List[st.SearchStrategy["Case"]]],
+) -> None:
+    """Gather all connections based on Open API links definitions."""
+    all_status_codes = list(endpoint.definition.resolved["responses"])
+    for status_code, target_endpoint, link in get_target_endpoints(endpoint):
+        strategy = make_links_strategy(
+            bundles[endpoint.verbose_name], target_endpoint, link, status_code, all_status_codes
+        )
+        sources[target_endpoint.verbose_name].append(strategy)
+
+
+def make_links_strategy(
+    bundle: Bundle, target_endpoint: "Endpoint", link: Dict[str, Any], status_code: str, all_status_codes: List[str]
+) -> st.SearchStrategy["Case"]:
+    """Create a single Open API Links based strategy.
+
+    It combines all links that point to an endpoint into a single strategy.
+    Only one link is used at once - responses are not combined with each other.
+    """
+    if status_code == "default":
+        response_filter = default_status_code(all_status_codes)
+    else:
+        response_filter = match_status_code(status_code)
+
+    target_strategy = target_endpoint.as_strategy()
+
+    def expand_parameter(parameter: str, expression: str) -> Tuple[Optional[str], str, str]:
+        try:
+            # The parameter name is prefixed with its location. Example: `path.id`
+            location, name = tuple(parameter.split("."))
+            return location, name, expression
+        except ValueError:
+            return None, parameter, expression
+
+    link_parameters = [
+        expand_parameter(parameter, expression) for parameter, expression in link.get("parameters", {}).items()
+    ]
+    link_request_body = link.get("requestBody", NOT_SET)
+
+    # Create a builder for strategies that will generate cases for the given endpoint
+    def make_strategy(item: Tuple["Case", requests.Response]) -> st.SearchStrategy["Case"]:
```

Comment:
Maybe it is better to create a special class, that holds all the needed context? and return its instance from `call_and_validate`?, then it will be easier to access its components